decidim-core/app/models/decidim/user.rb

```diff
@@ -53,6 +53,8 @@ def self.all
 
     has_one_attached :download_your_data_file
 
+    scope :marked_for_deletion, -> { where.not(marked_for_deletion_at: nil) }
+
```

Comment:
I see an alternative without the need of a migration by using solely what we already have:

```suggestion
    scope :inactive_users, -> { not_deleted.where(last_sign_in_at: ..Decidim.delete_inactive_users_after_days.ago).or(Decidim::User.not_deleted.where(last_sign_in_at: nil)) }

    scope :first_warning_inactive_users, -> { not_deleted.where(last_sign_in_at: ..Decidim.first_warning_inactive_users_after_days.ago).or(Decidim::User.not_deleted.where(last_sign_in_at: nil)) }

    scope :last_warning_inactive_users, -> { not_deleted.where(last_sign_in_at: ..Decidim.last_warning_inactive_users_after_days.ago).or(Decidim::User.not_deleted.where(last_sign_in_at: nil)) }
```